runtime/src/accounts_db.rs

```diff
@@ -6009,7 +6019,7 @@ impl AccountsDb {
             if pass == 0 {
                 // Need to add these last, otherwise older updates will be cleaned
                 for slot in &slots {
-                    self.accounts_index.add_root(*slot, false);
+                    self.add_root_to_update_index(*slot, false);
```

Comment:
We add any keys which have more than 1 key: https://github.com/solana-labs/solana/blob/d8984cb0f40eafbbf8aa2a95f3c0e43343e80230/runtime/src/accounts_db.rs#L5904-L5906

This uses the `uncleaned_pubkeys` and not `dirty_stores` which is why the `dirty_store` time is small, but I don't think we need to perform a clean on keys which only have 1 entry in them, no?

The other case is 0-lamport accounts. We can detect that in generate_index and also add it to the clean set.

Reply:
https://github.com/solana-labs/solana/pull/19041

Reply:
yeah, now i view things clearly. I totally agree with you.